contracts/examples/LiquidityExamples.sol

```diff
@@ -0,0 +1,249 @@
+        (tokenId, liquidity, amount0, amount1) = nonfungiblePositionManager.mint(params);
+
+        // Remove allowance and refund in both assets.
+        if (amount0 < amount0ToMint) {
+            TransferHelper.safeApprove(DAI, address(nonfungiblePositionManager), 0);
+            uint256 refund0 = amount0ToMint - amount0;
+            TransferHelper.safeTransfer(DAI, msg.sender, refund0);
+        }
+
+        if (amount1 < amount1ToMint) {
+            TransferHelper.safeApprove(USDC, address(nonfungiblePositionManager), 0);
+            uint256 refund1 = amount1ToMint - amount1;
+            TransferHelper.safeTransfer(USDC, msg.sender, refund1);
+        }
+    }
+
+    /// @notice Collects the fees associated with provided liquidity
+    /// @dev The contract must hold the erc721 token before it can collect fees
+    /// @param tokenId The id of the erc721 token
+    /// @return amount0 The amount of fees collected in token0
+    /// @return amount1 The amount of fees collected in token1
+    function collectAllFees(uint256 tokenId) external returns (uint256 amount0, uint256 amount1) {
+        // Caller must own the ERC721 position
+        // Call to safeTransfer will trigger `onERC721Received` which must return the selector else transfer will fail
+        nonfungiblePositionManager.safeTransferFrom(msg.sender, address(this), tokenId);
+
+        // set amount0Max and amount1Max to uint256.max to collect all fees
+        // alternatively can set recipient to msg.sender and avoid another transaction in `sendToOwner`
+        INonfungiblePositionManager.CollectParams memory params =
+            INonfungiblePositionManager.CollectParams({
+                tokenId: tokenId,
+                recipient: address(this),
+                amount0Max: type(uint128).max,
+                amount1Max: type(uint128).max
+            });
+
+        (amount0, amount1) = nonfungiblePositionManager.collect(params);
+
+        // send collected feed back to owner
+        _sendToOwner(tokenId, amount0, amount1);
+    }
+
+    /// @notice A function that decreases the current liquidity by half. An example to show how to call the `decreaseLiquidity` function defined in periphery.
+    /// @param tokenId The id of the erc721 token
+    /// @return amount0 The amount received back in token0
+    /// @return amount1 The amount returned back in token1
+    function decreaseLiquidityInHalf(uint256 tokenId) external returns (uint256 amount0, uint256 amount1) {
+        // caller must be the owner of the NFT
+        require(msg.sender == deposits[tokenId].owner, 'Not the owner');
+        // get liquidity data for tokenId
+        uint128 liquidity = deposits[tokenId].liquidity;
+        uint128 halfLiquidity = liquidity / 2;
+
+        // amount0Min and amount1Min are price slippage checks
+        // if the amount received after burning is not greater than these minimums, transaction will fail
```

Comment:
it actually might be fun to add a slippage check in the contract. say for instance decreaseLiquidityForStablePair. The liquidity amount could be a param or you could just do it in half again. This would be tricky though esp since some tokens like USDC uses 6 decimals -__- 

Ok this is actually prettty annoyingly tricky, and it might even merit its own example contract someday.

You'd have to find out token decimals from each token: `1e6` is 1USDC and `1e18` is 1DAI.
Then the price of 1 would be multiplied or divided by the difference in decimals depending on which token is token0.
Then you could use different libraries like: `LiquidityAmounts.getAmount0ForLiquidity()`and have a minimum that amount to get back which would signify how much of the price of 1 the current price is.

(not a blocker)

Reply:
I love this - yes we should definitely include this. Going to push the changes above for now so we can get the MVP guide up and will work on this as a next addition. Might need to run some questions by you so will leave this as unresolved.

Reply:
I can resolve for now. This feature is probably better for another time??

Reply:
yes. I'll work on this starting next week :)

Reply:
you can always get a second opinion on the idea too...other things may be a bigger priority